### PR TITLE
Give some love to public tracings

### LINKED ISSUE #1997: Error message on public tracings
Open in incognito browser: https://demo.webknossos.org/annotations/Explorational/599ad6f66c000070000ea174/readOnly#769,547,353,0,2.00,1
Error message: `You are not allowed to view this resource`

I suppose this is due to requests to `/api/scripts` or `/api/taskTypes`.

<a href="https://timer.scm.io/repos/537237ce1a00001a003c4a14/issues/1997/create?referer=github" target="_blank">Log Time</a>

### PR DESCRIPTION
**Pull request by daniel-wer**
Disable certain UI elements and hide permission error message for public tracings.

### Mailable description of changes:
 - Disabled some UI elements for public tracings, that would've resulted in permission errors on use.

### Steps to test:
- Share a tracing and make it publicly accessible. Access it when logged out (disable `enableDevAutoLogin` in `application.conf`). There should be no error message, the tracing should not be re-nameable, the share modal should not include the checkbox to make the tracing private again and the actions dropdown next to the read-only tag should only contain actions that make sense in the public tracing context.
- Nothing should've changed when logged in, except that read-only tracings should not be re-nameable now, which was a bug before.

### Issues:
- fixes #1997

------
- [x] Ready for review

### CODE REVIEW

**Comment by daniel-wer** [comment]
In my opinion `restrictions.allowFinish` should be `false` for public tracings that are accessed without being logged in. This would then automatically hide the `Archive` button in the actions dropdown. Waiting on @jfrohnhofens opinion on that :)

**Comment by jfrohnhofen** [comment]
@daniel-wer We use our default `readonlyAnnotation` restrictions for all read-only tracings (which includes public tracings). Right now those allow access, finish and download. I can change them to exclude finish, but that would also change things for read-only annotations. Is this a problem?

**Comment by daniel-wer** [comment]
I don't think this would be a problem. I don't know why for a read-only tracing someone would need to have permission to finish/archive the tracing.
@kevinbbb87 or @heikowissler, can you confirm that?

**Comment by kevinbbb87** [comment]
@daniel-wer 
I think you are right, that wouldn't be a problem

**Comment by daniel-wer** [comment]
Thanks @jfrohnhofen. Ready for review now.

File: app/models/annotation/AnnotationRestrictions.scala [hunk 1]
```diff
@@ -68,15 +68,12 @@ object AnnotationRestrictions {
...
-      override def allowFinish(user: Option[User]) = true
-      override def allowDownload(user: Option[User]) = true
     }
 
   def updateableAnnotation() =
     new AnnotationRestrictions {
       override def allowAccess(user: Option[User]) = true
       override def allowUpdate(user: Option[User]) = true
       override def allowFinish(user: Option[User]) = true
-      override def allowDownload(user: Option[User]) = true
```
Comment (hotzenklotz) [inline app/models/annotation/AnnotationRestrictions.scala]: Is `allowDownload` removed from all tracings / requests? If yes, please adjust the flow types accordingly.

Comment (jfrohnhofen) [reply app/models/annotation/AnnotationRestrictions.scala]: No, but `allowDownload` defaults to `allowAccess` by default. So we tend to omit it whenever the two are equal.

**Pull request merged**